src/test/java/org/terasology/minecarts/RailsTest.java

```diff
@@ -40,42 +44,58 @@
 
 @ExtendWith(MTEExtension.class)
 @Dependencies({"Rails", "CoreAssets"})
-@Disabled("Failing test with Reflection")
+@Tag("MteTest")
 public class RailsTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(RailsTest.class);
+
     private static final String RAIL_BLOCKFAMILY_URI = "rails:rails";
+
     private Block dirtBlock;
     private Block airBlock;
     private BlockFamily railBlockFamily;
 
-    @In
-    BlockEntityRegistry blockEntityRegistry;
+    private boolean initialized = false;
+
     @In
     BlockManager blockManager;
     @In
     ModuleTestingHelper helper;
     @In
     WorldProvider worldProvider;
 
-    public void initialize() {
+    public void setup() {
+        logger.info("Setup blocks and block families (should only happen once)");
         airBlock = blockManager.getBlock("engine:air");
         dirtBlock = blockManager.getBlock("CoreAssets:Dirt");
+        logger.info("✓ successfully resolved blocks");
         railBlockFamily = blockManager.getBlockFamily(RAIL_BLOCKFAMILY_URI);
+        logger.info("✓ successfully resolved block family");
+    }
 
-        BlockRegion region = new BlockRegion(0, 0, 0).expand(5, 5, 5);
+    @BeforeEach
+    public void initialize() {
+        logger.info("Pre-test initialization");
+        if (!initialized) {
+            initialized = true;
```

Comment:
I could be mixing things up with some other test framework. But as I understand it, `this.initialized` should _never_ be true here in this `@BeforeEach` function, because it'll be working with a new instance of RailsTest each time.

Reply:
the [logs](http://jenkins.terasology.io/teraorg/job/Terasology/job/Modules/job/R/job/Rails/job/PR-80/7/consoleFull) agree with me here. We see this for each of the tests:

[Test worker] INFO  org.terasology.minecarts.RailsTest - Pre-test initialization  
[Test worker] INFO  org.terasology.minecarts.RailsTest - Setup blocks and block families (should only happen once)

Reply:
Hm, then I have a wrong understanding of what is reused for the tests... 🤔 I'd like to use `@BeforeAll` here, but this requires static methods and fields...

Reply:
BeforeAll can be instance method, when TestInstanceLifeCycle is per-class. One test class instance per class

Or should be static otherwise. (TestInstanceLifeCycle per-method, test class instance per method )